src/components/partnership/dto/partnership.dto.ts

```diff
@@ -0,0 +1,39 @@
+import { GraphQLString } from 'graphql';
+import { Field, ObjectType } from 'type-graphql';
+import {
+  DateTimeField,
+  Resource,
+  SecuredDateTime,
+  SecuredProperty,
+} from '../../../common';
+import { Organization } from '../../organization';
+import { PartnershipAgreementStatus } from './partnership-agreement-status.enum';
+import { PartnershipType } from './partnership-type.enum';
+
+@ObjectType({
+  description: SecuredProperty.descriptionFor('a partnership agreement status'),
+})
+export abstract class SecuredPartnershipAgreementStatus extends SecuredProperty<
+  PartnershipAgreementStatus
+>(GraphQLString) {}
```

Comment:
Does this output the schema with the correct value type?
i.e.:
```graphql
type SecuredPartnershipAgreementStatus {
  value: PartnershipAgreementStatus
  ...
}
```

Reply:
I shamelessly lifted this approach from the Education service, and now upon further inspection I can see that it's returning a `value: String` in GQL. 

Here's my current approach:
```typescript
export abstract class SecuredPartnershipAgreementStatus extends SecuredProperty<
  PartnershipAgreementStatus
>(PartnershipAgreementStatus) {}
```

However, this is throwing an error: 
```
Argument of type 'typeof PartnershipAgreementStatus' is not assignable to parameter of type 'GraphQLScalarType | ClassType<PartnershipAgreementStatus> | AbstractClassType<PartnershipAgreementStatus>'.
  Type 'typeof PartnershipAgreementStatus' is not assignable to type 'AbstractClassType<PartnershipAgreementStatus>'.
    Property 'prototype' is missing in type 'typeof PartnershipAgreementStatus' but required in type '{ prototype: PartnershipAgreementStatus; }'.
```

It looks like `SecuredProperty` is expecting a value that can be a class, interface, or scalar. Enums don't seem to match any of these constraints. See `SecuredProperty` definition here:

```typescript
export function SecuredProperty<T>(
  ValueClass: ClassType<T> | AbstractClassType<T> | GraphQLScalarType
) {
  @ObjectType({ isAbstract: true, implements: [Readable, Editable] })
  abstract class SecuredPropertyClass
    implements Readable, Editable, Secured<T> {
    @Field(() => ValueClass, { nullable: true })
    readonly value?: T;
    @Field()
    readonly canRead: boolean;
    @Field()
    readonly canEdit: boolean;
  }

  return SecuredPropertyClass;
}
```

Any thoughts on how we can make `SecuredProperty` abstractly accept enum values while not compromising type safety in TS?

Reply:
Ok it looks like type-graphql emits schema correctly when passing the enum in, so it's just TS that doesn't like it.

My only thought is to add `object` to the acceptable input types.

Reply:
Perfect! That should do it!